### PR TITLE
Add support for last modified and extended status endpoints

### PR DESCRIPTION
**Pull request by DanWilliamsAu**
Adds support for a couple of endpoints that are not currently implemented in the SDK

* https://developer.smartling.com/v2.0/reference#get_projects-projectid-locales-localeid-file-status; and
* https://developer.smartling.com/v2.0/reference#get_projects-projectid-file-last-modified

### CODE REVIEW

File: file_status_extended.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	TotalStringCount      int
+	TotalWordCount        int
+}
+
+func (fs FileStatusExtended) AwaitingAuthorizationStringCount() int {
+	return fs.TotalStringCount - fs.AuthorizedStringCount - fs.ExcludedStringCount - fs.CompletedStringCount
+}
+
+func (fs FileStatusExtended) InProgressStringCount() int {
+	return fs.NotCompletedStringCount() - fs.AwaitingAuthorizationStringCount()
```
Comment (dimitrystd) [inline file_status_extended.go]: `fs.AuthorizedStringCount` this is what translator sees. So all authorized strings are "in progress" strings

File: file_status_extended.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+func (fs FileStatusExtended) NotCompletedStringCount() int {
+	return fs.TotalStringCount - fs.CompletedStringCount
```
Comment (dimitrystd) [inline file_status_extended.go]: What's the difference between `InProgressStringCount` and `NotCompletedStringCount`? Both sounds like - all authorized but not translated strings yet.

File: client_file_status_extended.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package smartling
+
+import "fmt"
+
+const (
+	endpointFileStatusExtended = "/files-api/v2/projects/%s/locales/%s/file/status"
+)
+
+// GetFileStatus returns extended file status for a file for a particular locale
+func (client *Client) GetFileStatusExtended(
```
Comment (dimitrystd) [inline client_file_status_extended.go]: Network overhead eats significant time. I would recommend to change your application code to do a single bulk status request (all locales at once) and then work with particular locales.

If you still need this method then i would like to ask to rename it `GetFileLocaleStatus`, and `FileStatusExtended` -> `FileLocaleStatus`

**Comment by DanWilliamsAu** [comment]
@dimitrystd thanks for your feedback.  I realise now that all the information I need is in fact on the regular status payload.  I've removed the extended endpoint code as you suggested.  I've also added a couple of helper methods to make it easier to work with.  Namely, a function to extract the FileStatusTranslation for a particular locale, and some methods for calculating awaiting authorisation numbers.

Let me know your thoughts!

**Pull request merged**